cloudplatform/connectivity-apache-httpclient5/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultApacheHttpClient5Factory.java

```diff
@@ -175,16 +170,22 @@ private RequestConfig getRequestConfig( @Nullable final HttpDestinationPropertie
 
     private boolean isProtocolUpgradeEnabled( @Nullable final HttpDestinationProperties destination )
     {
-        if( ENABLE_TLS_UPGRADE != null ) {
-            return ENABLE_TLS_UPGRADE;
-        }
-        if( destination == null ) {
-            return true;
-        }
-        if( destination.getTlsVersion().isDefined() ) {
-            return false;
+        switch( tlsUpgrade ) {
+            case ENABLED:
+                return true;
+            case DISABLED:
+                return false;
+            case INTERNET:
+                if( destination == null ) {
+                    return true;
+                }
+                if( destination.getTlsVersion().isDefined() ) {
+                    return false;
+                }
+                return !destination.getProxyType().contains(ProxyType.ON_PREMISE);
+            default:
+                throw new IllegalStateException("Unknown TLS upgrade setting: " + tlsUpgrade);
```

Comment:
Thanks to Java 17, can be simplified to:

```suggestion
        return switch( tlsUpgrade ) {
            case ENABLED -> true;
            case DISABLED -> false;
            case INTERNET -> {
                if( destination == null ) {
                    yield true;
                }
                if( destination.getTlsVersion().isDefined() ) {
                    yield false;
                }
                yield !destination.getProxyType().contains(ProxyType.ON_PREMISE);
            }
        };
```

Reply:
yield... :o